lib/internal/backend/repository/repository.go

```diff
@@ -108,27 +108,27 @@ func (rb *RepositoryBackend) BuildACI(layerIDs []string, dockerURL *types.Parsed
 	}
 }
 
+// checkRegistryStatus determines registry API version compatibility according to spec:
+// https://docs.docker.com/registry/spec/api/#/api-version-check
 func checkRegistryStatus(statusCode int, hdr http.Header, version registryVersion) (bool, error) {
 	switch statusCode {
 	case http.StatusOK, http.StatusUnauthorized:
 		ok := true
 		if version == registryV2 {
-			// v2 API requires this check
+			// According to v2 spec, registries SHOULD set this header value
+			// and clients MAY fallback to v1 if missing, as done here.
 			ok = hdr.Get("Docker-Distribution-API-Version") == "registry/2.0"
 		}
 		return ok, nil
-	case http.StatusNotFound:
+	default:
 		return false, nil
 	}
-
-	return false, fmt.Errorf("unexpected http code: %d", statusCode)
```

Comment:
Local build+test seems fine (go 1.6.3), but I see over-zealous checkers may get confused. I'll switch to a false-return at the bottom.